fdbserver/DataDistribution.actor.cpp

```diff
@@ -587,6 +692,14 @@ struct DDTeamCollection {
 		return result && resultEntries.size() == 0;
 	}
 
+	// Used for map<Reference<TCMachineTeamInfo>, int>
+	struct CompareTCMachineTeamInfoRef {
+		bool operator() (const Reference<TCMachineTeamInfo> &lhs, const Reference<TCMachineTeamInfo> &rhs) const {
+			return lhs->machineIDs < rhs->machineIDs;
+		}
+	};
```

Comment:
There's not really a need to write code in this style anymore in C++, as this was the workaround for the lack of lambdas.

    auto CompareTCMachineTeamInfoRef = []( ... ) -> bool { return ...; }

Is now the idiomatic way to write this, and you can just inline the lambda into the call site if you only need it once.

It's also worth considering if you should just specialize std::less<> for the type you need.